docs/configuration/node_affinity.md

```diff
@@ -0,0 +1,192 @@
+# Node Affinity & Cluster Topology
+
+## Node Affinity
+
+Kubernetes allows pods to be assigned to nodes based on various critera through [node affinity][k8s-node-affinity].
+
+M3DB was built with failure tolerance as a core feature. M3DB's [isolation groups][m3db-isogroups] allow shards to be
+placed across failure domains such that the loss of no single domain can cause the cluster to lose quorum. More details
+on M3DB's resiliency can be found in the [deployment docs][m3db-deployment].
+
+By leveraging Kubernetes' node affinity and M3DB's isolation groups, the operator can guarantee that M3DB pods are
+distributed across failure domains. For example, in a Kubernetes cluster spread across 3 zones in a cloud region, the
+`isolationGroups` below config would guarantee that no single zone failure could degrade the M3DB cluster.
+
+M3DB is unaware of the underlying zone topology: it just views the isolation groups as `group1`, `group2`, `group3` in
+its [placement][m3db-placement]. Thanks to the Kubernetes scheduler, however, these groups are actually scheduled across
+separate failure domains.
+
+```yaml
+apiVersion: operator.m3db.io/v1alpha1
+kind: M3DBCluster
+...
+spec:
+  replicationFactor: 3
+  isolationGroups:
+  - name: group1
+    numInstances: 3
+    nodeAffinityTerms:
+    - key: failure-domain.beta.kubernetes.io/zone
+      values:
+      - us-east1-b
+  - name: group2
+    numInstances: 3
+    nodeAffinityTerms:
+    - key: failure-domain.beta.kubernetes.io/zone
+      values:
+      - us-east1-c
+  - name: group3
+    numInstances: 3
+    nodeAffinityTerms:
+    - key: failure-domain.beta.kubernetes.io/zone
+      values:
+      - us-east1-d
+```
+
+## Tolerations
+
+In addition to allowing pods to be assigned to certain nodes via node affinity, Kubernetes allows pods to be _repelled_
+from nodes through [taints][k8s-taints] if they don't tolerate the taint. For example, the following config would ensure:
+
+1. Pods are spread across zones.
+
+2. Pods are only assigned to nodes in the `m3db-dedicated-pool` pool.
+
+3. No other pods could be assigned to those nodes (assuming they were tainted with the taint `m3db-dedicated-taint`).
+
+```yaml
+apiVersion: operator.m3db.io/v1alpha1
+kind: M3DBCluster
+...
+spec:
+  replicationFactor: 3
+  isolationGroups:
+  - name: group1
+    numInstances: 3
+    nodeAffinityTerms:
+    - key: failure-domain.beta.kubernetes.io/zone
+      values:
+      - us-east1-b
+    - key: nodepool
+      values:
+      - m3db-dedicated-pool
+  - name: group2
+    numInstances: 3
+    nodeAffinityTerms:
+    - key: failure-domain.beta.kubernetes.io/zone
+      values:
+      - us-east1-c
+    - key: nodepool
+      values:
+      - m3db-dedicated-pool
+  - name: group3
+    numInstances: 3
+    nodeAffinityTerms:
+    - key: failure-domain.beta.kubernetes.io/zone
+      values:
+      - us-east1-d
+    - key: nodepool
+      values:
+      - m3db-dedicated-pool
+  tolerations:
+  - key: m3db-dedicated
```

Comment:
Should this need to be `m3db-dedicated-pool` or is `m3db-dedicated` and `m3db-dedicated-pool` somehow related somewhere?

Reply:
So `m3db-dedicated-pool` the nodepool and `m3db-dedicated` the taint are two separate things, I wanted to convey that they could be named differently (like `foo` and `bar` even). Maybe I should make them even more different to clarify that? Or can just mention it